### DISCUSSION TITLE
NullReferenceException running async rules in 5.3.0

### DISCUSSION THREAD

**Discussion by ajohnstone-ks** [MarimerLLC/csla#1768] [Questions] [answered]
I'm trying to repo this outside our main solution, but haven't had much luck so far, so I thought I'd ask here.

After running an async rule, which does a remote data portal call to load additional information from the database, there's an exception being thrown `BusinessRules.RunRule`s as it sets `RunningAsyncBusinessRules = false`.

The issue is that `busyChanged` (the `AsyncManualResetEvent`) is null.  I see this is initialized when the object is constructed, so I'm not sure how it would happen.

Any ideas?

**Comment by rockfordlhotka** [comment, marked as answer]
I think this is the same issue? https://github.com/MarimerLLC/csla/issues/1746

**Reply by ajohnstone-ks** [reply]
That looks very close; we are using `BinaryFormatter` still.  We're not getting it during the fetch though, that works fine, it happens when the UI code actually sets the property which has an async rule associated with it.
